Source/WebCore/loader/PolicyChecker.cpp

```diff
@@ -76,19 +77,11 @@ static bool isAllowedByContentSecurityPolicy(const URL& url, const Element* owne
     return ownerElement->document().contentSecurityPolicy()->allowChildFrameFromSource(url, redirectResponseReceived);
 }
 
-PolicyCheckIdentifier PolicyCheckIdentifier::create()
+static bool shouldExecuteJavaScriptURLSynchronously(const URL& url)
 {
-    static uint64_t identifier = 0;
-    identifier++;
-    return PolicyCheckIdentifier { Process::identifier(), identifier };
-}
-
-bool PolicyCheckIdentifier::isValidFor(PolicyCheckIdentifier expectedIdentifier)
-{
-    RELEASE_ASSERT_WITH_MESSAGE(m_policyCheck, "Received 0 as the policy check identifier");
-    RELEASE_ASSERT_WITH_MESSAGE(m_process == expectedIdentifier.m_process, "Received a policy check response for a wrong process");
-    RELEASE_ASSERT_WITH_MESSAGE(m_policyCheck <= expectedIdentifier.m_policyCheck, "Received a policy check response from the future");
-    return m_policyCheck == expectedIdentifier.m_policyCheck;
+    // FIXME: Some sites rely on the javascript:'' loading synchronously, which is why we have this special case.
+    // Blink has the same workaround (https://bugs.chromium.org/p/chromium/issues/detail?id=923585).
+    return url == "javascript:''"_s || url == "javascript:\"\""_s;
```

Comment:
Should this ignore ASCII case?

Reply:
I am merely moving the code. This is a quirk and Blink seems to be doing a case sensitive check like us:
https://chromium-review.googlesource.com/c/chromium/src/+/1435821/